examples/src/main/java/io/grpc/examples/errorhandling/ErrorHandlingClient.java

```diff
@@ -0,0 +1,220 @@
+import io.grpc.examples.helloworld.HelloRequest;
+import io.grpc.stub.StreamObserver;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nullable;
+
+/**
+ * Shows how to extract error information from a server response.
+ */
+public class ErrorHandlingClient {
+  public static void main(String [] args) throws Exception {
+    new ErrorHandlingClient().run();
+  }
+
+  private Server server;
+  private ManagedChannel channel;
+
+  void run() throws Exception {
+    server = ServerBuilder.forPort(0).addService(new GreeterGrpc.AbstractGreeter() {
+      @Override
+      public void sayHello(HelloRequest request, StreamObserver<HelloReply> responseObserver) {
+        responseObserver.onError(Status.INTERNAL
+            .withDescription("Eggplant Xerxes Crybaby Overbite Narwhal").asRuntimeException());
+      }
+    }).build().start();
+    channel =
+        ManagedChannelBuilder.forAddress("localhost", server.getPort()).usePlaintext(true).build();
+
+    blockingCall();
+    futureCallDirect();
+    futureCallCallback();
+    asyncCall();
+    advancedAsyncCall();
+
+    channel.shutdown();
+    server.shutdown();
+    channel.awaitTermination(1, TimeUnit.SECONDS);
+    server.awaitTermination();
+  }
+
+  void blockingCall() {
+    GreeterBlockingStub stub = GreeterGrpc.newBlockingStub(channel);
+    try {
+      stub.sayHello(HelloRequest.newBuilder().setName("Bart").build());
+    } catch (StatusRuntimeException e) {
+      Status status = e.getStatus();
+      Verify.verify(status.getCode() == Status.Code.INTERNAL);
+      Verify.verify(status.getDescription().contains("Eggplant"));
+      // Cause is not transmitted over the wire.
+    }
+  }
+
+  void futureCallDirect() {
+    GreeterFutureStub stub = GreeterGrpc.newFutureStub(channel);
+    ListenableFuture<HelloReply> response =
+        stub.sayHello(HelloRequest.newBuilder().setName("Lisa").build());
+
+    try {
+      response.get();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(e);
+    } catch (ExecutionException e) {
+      StatusRuntimeException sre = (StatusRuntimeException) e.getCause();
+      Status status = sre.getStatus();
+      Verify.verify(status.getCode() == Status.Code.INTERNAL);
+      Verify.verify(status.getDescription().contains("Xerxes"));
+      // Cause is not transmitted over the wire.
+    }
+  }
+
+  void futureCallCallback() {
+    GreeterFutureStub stub = GreeterGrpc.newFutureStub(channel);
+    ListenableFuture<HelloReply> response =
+        stub.sayHello(HelloRequest.newBuilder().setName("Maggie").build());
+
+    final CountDownLatch latch = new CountDownLatch(1);
+
+    Futures.addCallback(response, new FutureCallback<HelloReply>() {
+      @Override
+      public void onSuccess(@Nullable HelloReply result) {
+        // Won't be called.
```

Comment:
Make it clear it won't be called because the server always fails in this example, not because that's how the API works.